academiverse-api/src/main/java/com/academiverse/academiverse_api/service/EnrolmentService.java

```diff
@@ -0,0 +1,80 @@
+package com.academiverse.academiverse_api.service;
+
+import com.academiverse.academiverse_api.dto.response.BaseResponse;
+import com.academiverse.academiverse_api.model.Enrolment;
+import com.academiverse.academiverse_api.model.Instruct;
+import com.academiverse.academiverse_api.model.User;
+import com.academiverse.academiverse_api.repository.EnrolmentRepository;
+import com.academiverse.academiverse_api.repository.InstructRepository;
+import com.academiverse.academiverse_api.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+public class EnrolmentService {
+    private final EnrolmentRepository enrolmentRepository;
+    private final InstructRepository instructRepository;
+    private final UserRepository userRepository;
+
+    public BaseResponse<List<Long>> getEligibleStudents(Long instructId) {
+        Optional<Instruct> instruct = instructRepository.findById(instructId);
+        BaseResponse<List<Long>> response = new BaseResponse<>();
+
+        if (instruct.isPresent()) {
+            // Retrieve the department ID from the course
+            Long departmentId = instruct.get().getCourse().getDepartment().getDepartmentId();
+
+            // Get all users
+            List<User> allUsers = userRepository.findAll(); // Fetch all users
+
+            // Get a list of enrolled students
+            List<Long> enrolledStudents = enrolmentRepository.findByInstructId(instructId)
+                    .stream()
+                    .map(Enrolment::getUserId)
+                    .collect(Collectors.toList());
+
+            // Filter eligible students based on department and enrollment status
+            List<Long> eligibleStudents = allUsers.stream()
```

Comment:
Logic seems little wrong. **!enrolledStudents.contains(user.getUserId())** this will give you students which are not enrolled. But we want enrolled students + unenrolled students from same department. Also the API response will have List of userId which won't be useful at frontend. We require whole user data including a flag to mark user whether it is already enrolled or unenrolled. @Pari4113